src/controllers/auth.controller.js

```diff
@@ -0,0 +1,269 @@
+const { User } = require('../models/user.js');
+const userService = require('../services/user.service.js');
+const jwtService = require('../services/jwt.service.js');
+const ApiError = require('../exeptions/api.error');
+const bcrypt = require('bcrypt');
+const tokenService = require('../services/token.service.js');
+const sendResetEmail = require('../services/resetEmail.services.js');
+const sendConfirmationEmail = require('../services/sendConfirmationEmail.js');
+// const { user } = require("../../src copy/models/user");
+
+function validateEmail(value) {
+  if (!value) {
+    return 'Email is required';
+  }
+
+  const emailPattern = /^[\w.+-]+@([\w-]+\.){1,3}[\w-]{2,}$/;
+
+  if (!emailPattern.test(value)) {
+    return 'Email is not valid';
+  }
+}
+
+function validatePassword(value) {
+  if (!value) {
+    return 'Password is required';
+  }
+
+  if (value.length < 6) {
+    return 'At least 6 characters';
+  }
+}
+
+const register = async (req, res, next) => {
+  const { name, email, password } = req.body;
+
+  const errors = {
+    email: validateEmail(email) || null,
+    password: validatePassword(password) || null,
+  };
+
+  if (errors.email || errors.password) {
+    throw ApiError.badRequest('Bad request', errors);
```

Comment:
Consider using `return next(ApiError.badRequest('Bad request', errors));` instead of `throw` to ensure the error is passed to the next middleware for consistent error handling.